### ISSUE TITLE
conflict resolution logic not working for address objects

### ISSUE THREAD

**Issue by Grisha-L** [echickering/panos-to-scm#11]
We've moved some configuration to SCM a few days back.
Today we tried to run the code again, with new and modified object.
The code properly identifies the conflict between XML and SCM, it requests to decide how to handle the conflicts, but neither Replace nor Merge doing the job, the object remains with the old value.

`2024-08-22 13:34:41,110 - INFO - Fetching items: https://api.sase.paloaltonetworks.com/sse/config/v1/addresses?limit=20000&position=&folder=All
Conflict detected for object 'object-rk-02'
SCM Object: {'ip_netmask': '10.20.30.40', 'id': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'name': 'object-rk-02'}
New Object: {'ip_netmask': '10.20.30.50', 'id': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'name': 'object-rk-02'}
Choose an action: [M]erge, [R]eplace, [C]reate new with appended name, [I]gnore (M/R/C/I): R
2024-08-22 13:34:54,804 - INFO - Fetching items: https://api.sase.paloaltonetworks.com/sse/config/v1/addresses?folder=All
2024-08-22 13:34:56,213 - ERROR - Current object 'object-rk-02' not found in SCM. Skipping update.`

**Comment by echickering** [comment]
Unable to reproduce here

2024-08-22 11:48:37,511 - INFO - Fetching items: https://api.sase.paloaltonetworks.com/sse/config/v1/addresses?limit=20000&position=&folder=All
Conflict detected for object 'H-10.255.255.1-remove'
SCM Object: {'ip_netmask': '10.255.255.111', 'name': 'H-10.255.255.1-remove', 'id': 'd2e4304a-3ab6-4025-bcb5-2eb84cbedacf'}
New Object: {'ip_netmask': '10.255.255.1', 'name': 'H-10.255.255.1-remove', 'id': 'd2e4304a-3ab6-4025-bcb5-2eb84cbedacf'}
Choose an action: [M]erge, [R]eplace, [C]reate new with appended name, [I]gnore (M/R/C/I): R
2024-08-22 11:48:39,792 - INFO - Fetching items: https://api.sase.paloaltonetworks.com/sse/config/v1/addresses?folder=All
2024-08-22 11:48:40,349 - INFO - Updating Address: H-10.255.255.1-remove at endpoint: /sse/config/v1/addresses/d2e4304a-3ab6-4025-bcb5-2eb84cbedacf
2024-08-22 11:48:41,187 - INFO - Updated Address: H-10.255.255.1-remove with values: {'name': 'H-10.255.255.1-remove', 'tag': [], 'ip_netmask': '10.255.255.1'}

**Comment by Grisha-L** [comment]
so I took a clean tenant for testing, and I was struggling to reproduce it as well...
Eventually I was able to identify that a very specific change event is causing those "replace" action to fail with any object.

1. duplicate an existing object
2. change the value of the original object
3. run the script

from now on, operation of adjusting a value of any object will fail.

please let me know if you were able to reproduce it as well.

**Comment by echickering** [comment]
What do you mean by 'duplicate' an existing object?

**Comment by Grisha-L** [comment]
Hey, I thought I figured out the reaction chain that led to the error, but
apparently there are must be more conditions, I'll post here when I'll be
sure how to reproduce the error on a clean tenant

**Comment by Grisha-L** [comment]
ok, I found the issue, there is a line of code in "panos-to-scm/scm/process.py", line 255-257
`                params = {scope_type: scope_value}
                
                current_objects = self.api_handler.get(entry_class.get_endpoint(), params=params)`
                
The params are missing definition of "limit", which apparently by default is 200 (I guess it's defined in SCM itself as I was unable to find such default in your code).
My env' has over 4k objects, hence it was scanning only small portion of the possible objects when looking up for the actual replace action, so it was a matter of luck.
Adding the limit manually did solve the issue for me.
I've created a pull request with the fix: https://github.com/PaloAltoNetworks/panos-to-scm/pull/15
